optimum/exporters/openvino/model_patcher.py

```diff
@@ -1493,41 +1493,69 @@ def _phi3_self_attn_sdpa_forward(
     return attn_output, None, past_key_value
 
 
-class Phi3ModelPatcher(OVDecoderModelPatcher):
-    def __enter__(self):
-        super().__enter__()
+# Adapted from https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/models/phi3/modeling_phi3.py#L324
+# and https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/modeling_rope_utils.py#L30
+def _phi3_longrope_forward(self, x, position_ids):
+    """
+    LongRoPE uses different scaling factors (short_factor vs long_factor) depending on whether
+    the sequence length exceeds the original max position embeddings used during pretraining.
+
+    Note: In transformers, the @dynamic_rope_update decorator replaces self.inv_freq before the forward pass.
+    Here we use torch.where to select between inv_freq and long_inv_freq and add the selection logic into the model graph.
+    """
+    seq_len = torch.max(position_ids) + 1
```

Comment:
I don't see that `short_factor` is extracted and used anyhow in this patch.
Please check reference impl.: https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_rope_utils.py#L454C5-L454C18

We need to be aligned with HF

Reply:
short_factor is not used explicitly because inv_freq is used directly. inv_freq is set here https://github.com/huggingface/transformers/blob/v4.55.1/src/transformers/models/phi3/modeling_phi3.py#L313C1-L314C69 . It calls compute_longrope_parameters in https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_rope_utils.py . It is called without specifying a seq_len, so in line 477 ext_factors is set using the short_factor (from the model config) and in line 479 inv_freq is set based on these ext_factors.

This code with explicit short and long factors is needed for transformers because they use this for inference, but for model export inv_freq will be set with short factor correctly.

Reply:
We are using this code during model export, as part of model loading code. During model loading, compute_longrope_embeddings is actually called with seq_len=None, so self.inv_freq will be set with short_factor (see your screenshot: if seq_len is None,  ext_factors is set according to short_factor, which is defined earlier in the function from model config, and inv_freq is than computed based on this short_factor. So, self.inv_freq is always set with short factor, self.long_inv_freq with long factor, and then in the _phi3_longrope_forward inv_freq is set to self.long_inv_freq if seq is long and else to self.inv_freq. This is aligned with transformers.